#### <a id="result-page"></a>Result Page

![ProtVar Result Page](images/result_page.png)
**A.** [Downloads](#further-info:downloads) A. My downloads contains all the files you have submitted for download. These can be downloaded from here to your local computer. Users can return to the same results again by clicking the 'Results' tab. The results will remain during the users session or until they are replaced by new results.   
**B.** Navigate to the previous or next page of results and choose how many results to display on the page.
**C.** The legend shows the colour schemes and category boundaries for scores and predictions. The colours can be switched between ProtVar standardised colours and the original colours.
**D.** Download all variant results. The file will be available in [Downloads](#further-info:downloads)  
**E.** The data is split into [genomic position](#further-info:genomic-position) and [protein position](#further-info:protein-position) and [annotations](#further-info:annotations)  
**F.** The types and number of user inputs and processing by ProtVar is shown here.
**G.** The [data type](#further-info:data-types) used to submit the variant is shown in pink. Multiple lines may be shown for a single input in cases such as overlapping genes or when the specific SNP is ambiguous.  
**H.** [CADD](#further-info:cadd) score is a nucleotide level predictor of pathogenicity.  
**I.** Mappings to different [isoforms](#further-info:isoforms) can be observed by expanding.  
**J.** AlphaMissense is an amino acid level predictor of pathogenicity.
**K.** Each of three Annotation types [Function annotation](#further-info:function-annotations) and [Population Observations](#further-info:population-observations) and [Structure Annotations](#further-info:structure-annotations) regarding each variant can be explored by clicking on the icons.
